src/components/Profile/components/PersonalInfoList/UpdateUserPrivacyDialog/index.jsx

```diff
@@ -0,0 +1,80 @@
+import { FormControl, IconButton, Input, InputLabel } from '@mui/material';
+import GordonSnackbar from 'components/Snackbar';
+import { useState, useEffect } from 'react';
+import userService from 'services/user';
+import SearchField from 'views/PeopleSearch/components/SearchFieldList/components/SearchField';
+
+const UpdateUserPrivacy = (username, field) => {
+  const [snackbar, setSnackbar] = useState({ message: '', severity: null, open: false });
+  const [group, setGroup] = useState('');
+  const [groupList, setGroupList] = useState([]);
+  let visibleTo = '';
+
+  const handlePrivacy = async (event) => {
+    try {
+      await userService.setUserPrivacy({ Field: field, VisibilityGroup: event.target.value });
+      setSnackbar({
+        message: 'Your office location will update within a couple hours.',
+        severity: 'success',
+        open: true,
+      });
+    } catch {
+      setSnackbar({
+        message: 'Office location failed to update. Please contact CTS.',
+        severity: 'error',
+        open: true,
+      });
+    }
+  };
+
+  useEffect(() => {
+    userService.getVisibilityGroups().then(setGroupList);
+  }, []);
+
+  useEffect(() => {
+    userService.getPrivacySetting(username).then(setGroup);
+  }, [group]);
+
+  let tempField = field;
+  if (field == 'HomeCity HomeState') {
+    field = 'HomeCity';
+  } else if (field == 'Country HomeCountry') {
+    field = 'Country';
```

Comment:
```suggestion
  if (field == 'HomeCity HomeState') {
    field = 'HomeCity';
  } else if (field == 'Country HomeCountry') {
    field = 'Country';
```
```suggestion
  if (field == 'HomeCity HomeState') {
    field = 'HomeCity';
  } else if (field == 'Country HomeCountry') {
    field = 'Country';
```

why are fields separated by space?

Reply:
Same reason as what I comments in API:

"This is because some fields are controlled by the same button, such as Country and HomeCountry. Whenever an user choose the Visibility for Home, both Country and HomeCountry changed. Thus. I pass 'Country HomeCountry' as Field in this case and need to split it by space. (Might not be a good practice but not sure if there is an easy way to improve it.."